File: client/src/types/JsonModels/Users/AdminUserModel.ts
import type { EntitlementSource } from "../Enums/EntitlementSource";

export interface AdminUserModel
{
	id: number;
	email: string;
	firstName?: string;
	lastName?: string;
	isActive: boolean;
	isAdmin: boolean;
	dateCreated: string;
	lastLoginAt?: string;
	effectivePlanCode: string;
	effectivePlanName: string;
	source: EntitlementSource;
	hasActiveOverride: boolean;
}

File: server/FitMate.Core/JsonModels/Users/AdminUserModel.cs
namespace FitMate.Core.JsonModels.Users;

public class AdminUserModel
{
    public long Id { get; set; }
    public string Email { get; set; } = string.Empty;
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public bool IsActive { get; set; }
    public bool IsAdmin { get; set; }
    public DateTime DateCreated { get; set; }
    public DateTime? LastLoginAt { get; set; }
}
